### PR TITLE
Exclude namespace from PolicyRecommendationRegistrar and fix some tests

### PR DESCRIPTION
**Pull request by deefreak**
1. The PR has exclude namespace logic from PolicyRecommendationRegistrar.
2. Changes in Makefile to print code coverage for the tests.
3. Some other fixes in the test files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,7 +59,8 @@ test: prometheus manifests generate fmt vet envtest ## Run tests with Prometheus
 	@echo "Starting Prometheus for testing..."
 	@$(PROMETHEUS) --config.file=pkg/testconfig/prometheus.yml & \
 	trap 'killall prometheus; rm -rf data' EXIT; \
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" ginkgo run -p 1 ./... --coverprofile cover.out -test.v --procs 1 --vv --output-interceptor-mode=none
+    KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" ginkgo run --cover --coverprofile cover.out ${OTTOSCALR_TEST_ARGS} --procs 1 --nodes 1 -v --output-interceptor-mode=none ./... -test.v
```
Comment (bharathguvvala) [inline Makefile:62]: you can remove the -test.v flag at the end

File: pkg/controller/policyrecommendation_registrar.go [hunk 2]
```diff
@@ -206,6 +208,21 @@ func (controller *PolicyRecommendationRegistrar) SetupWithManager(mgr ctrl.Manag
 		},
 	}
 
+	includeNamespacePredicate := predicate.Funcs{
```
Comment (bharathguvvala) [inline pkg/controller/policyrecommendation_registrar.go:211]: change this to just namespaceFilter

File: pkg/controller/policyrecommendation_registrar.go [hunk 3]
```diff
@@ -239,5 +256,15 @@ func (controller *PolicyRecommendationRegistrar) SetupWithManager(mgr ctrl.Manag
 			},
 			builder.WithPredicates(deletePredicate),
 		).
+		WithEventFilter(includeNamespacePredicate).
 		Complete(controller)
 }
+
+func (controller *PolicyRecommendationRegistrar) isIncludedNamespace(namespace string) bool {
```
Comment (bharathguvvala) [inline pkg/controller/policyrecommendation_registrar.go:263]: rename this to isWhitelistedNamespace

**Review by bharathguvvala — APPROVED** [review]
Some minor changes requested for readability.

File: pkg/controller/policyrecommendation_registrar.go (hunk 3 above)
Comment (deefreak) [reply pkg/controller/policyrecommendation_registrar.go]: Done

File: pkg/controller/policyrecommendation_registrar.go (hunk 2 above)
Comment (deefreak) [reply pkg/controller/policyrecommendation_registrar.go]: Done

File: Makefile (hunk 1 above)
Comment (deefreak) [reply Makefile]: done

**Pull request merged**